scripts/run-reset-hammer.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Copyright (c) 2025 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+BOARD=$1
+COUNT=$2
+
+if [ $# -ne 2 ]; then
+	echo "Incorrect number of arguments, expected board name and test count"
+	exit 1
+fi
+
+# Repeatedly resets the SMC. Intended to be run within CI.
+# Accepts two arguments:
+# 1. board name to use
+# 2. the number of times to run the reset test
+
+TT_Z_P="$(dirname $(realpath $0))/.."
+echo $TT_Z_P
+
+# Build the test
+west build -p always -b $BOARD -d build-reset-test $TT_Z_P/app/smc
+
+for i in $(seq 1 $COUNT); do
+	echo "############ Iteration $i ##################"
+	# Flash twice, this gives us the best chance of resetting the SMC while
+	# I2C init is ongoing
+	west flash -d build-reset-test
+	west flash -d build-reset-test
+	# Rescan PCI
+	sudo $TT_Z_P/scripts/rescan-pcie.sh
+	# Test with tt-smi
+	tt-smi -s
+done
```

Comment:
It would be ideal if we could actually count and report how many times the process failed vs succeeded (even if the expectation is that it would always succeed).

Also, be careful about running with `set -e`. If any command has a non-zero exit status, then the script will die prematurely.

It would also be good to run the script through `shellcheck`. It will point out bashisms, so there are differences between running with `#!/bin/sh` vs `#!/bin/bash`.